platform/internal/SysTimer.h

```diff
@@ -32,12 +32,8 @@ namespace mbed {
 namespace internal {
 
 /**
- * @cond MBED_INTERNAL
- *
- * @addtogroup mbed
- * @{
- *
- * @defgroup mbed_SysTimer SysTimer class
+ * \defgroup mbed_SysTimer SysTimer class
```

Comment:
I believe this file is not included by any public header, so can go into `source`. (The logic being that the source directory contains all source and headers that could be removed from a binary-only distribution, or not be on the include path at all for stuff outside the Mbed OS "source" directories).

Same for mbed_os_timer.h

Except in both cases they're being accessed from a _different_ source directory to this one. Not sure if that forces them to be outside this source directory. Depends how we imagine future path configuration would look.

Reply:
`SysTimer.h` is included by `mbed_os_timer.cpp` & `mbed_os_timer.cpp` - both in `platform` so no problem at all to move to `platform/source`

`mbed_os_timer.h` is not included by any public header but by a few source files in `rtos` and `platform`. I would move it under `platform/source` for now but indeed, we might have to consolidate `platform` & `rtos` in the future or decide if headers accessed by different source directories should be moved to `internal`

Reply:
I thought I saw one reference over in `events`.

If this was Yotta, then cross-component API would have to be in `platform/internal`, but we can figure out if we need that restriction later as you say.